tests/io/v3/test_groups.py

```diff
@@ -0,0 +1,142 @@
+"""
+Testing the Groups endpoints
+"""
+import responses
+from responses import matchers
+
+GROUPS_BASE_URL = r"https://cloud.tenable.com/api/v3/groups"
+BASE_URL = r"https://cloud.tenable.com"
+GROUP_ID = "b27778c1-af10-4218-a1a4-c2c36b236e05"
+USER_ID = "e9f23194-adb7-4c02-8632-615c694c787e"
+
+
+@responses.activate
+def test_add_user(api):
+    responses.add(
+        responses.POST,
+        f"{GROUPS_BASE_URL}/{GROUP_ID}/users/{USER_ID}"
+    )
+    assert None is api.v3.groups.add_user(GROUP_ID, USER_ID)
+
+
+@responses.activate
+def test_create(api):
+    group_name = "Test Group"
+    responses.add(
+        responses.POST,
+        f"{GROUPS_BASE_URL}",
+        match=[matchers.json_params_matcher({"name": group_name})],
+        json={
+            "id": GROUP_ID,
+            "name": group_name,
+        },
+    )
+    resp = api.v3.groups.create(group_name)
+    assert group_name == resp["name"]
+    assert GROUP_ID == resp["id"]
+
+
+@responses.activate
+def test_delete(api):
+    responses.add(responses.DELETE, f"{GROUPS_BASE_URL}/{GROUP_ID}")
+    assert None is api.v3.groups.delete(GROUP_ID)
+
+
+@responses.activate
+def test_delete_user(api):
+    responses.add(
+        responses.DELETE,
+        f"{GROUPS_BASE_URL}/{GROUP_ID}/users/{USER_ID}"
+    )
+    assert None is api.v3.groups.delete_user(GROUP_ID, USER_ID)
+
+
+@responses.activate
+def test_edit(api):
+    updated_group_name = "Updated Test Group"
+
+    responses.add(
+        responses.PUT,
+        f"{GROUPS_BASE_URL}/{GROUP_ID}",
+        match=[matchers.json_params_matcher({"name": updated_group_name})],
+        json={"id": GROUP_ID, "name": updated_group_name, "user_count": 0},
+    )
+
+    resp = api.v3.groups.edit(GROUP_ID, updated_group_name)
+    assert GROUP_ID == resp["id"]
+    assert updated_group_name == resp["name"]
+
+
+@responses.activate
+def test_list(api):
```

Comment:
Can we add a test case for search as well and skip it for now which can be updated later once the search method is implemented in API. otherwise, the changes look good to merge.

Reply:
Done.